MatrixSDK/Utils/Profiling/MXTaskProfile.h

```diff
@@ -0,0 +1,38 @@
+// 
+// Copyright 2020 The Matrix.org Foundation C.I.C
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface MXTaskProfile : NSObject
+
+@property (nonatomic, readonly) NSString *name;
+@property (nonatomic, readonly) NSString *category;
+
+@property (nonatomic, readonly) NSDate *startDate;
+@property (nonatomic, readonly, nullable) NSDate *endDate;
+@property (nonatomic, readonly) NSTimeInterval duration;
+
+@property (nonatomic) NSUInteger units;
```

Comment:
Can we have more description about what this field is intended to carry?
Maybe we can add `metadata: NSDictionary` field even if Matomo does not support at the moment. We can use known dictionary keys like: `count` and let the possibility to set custom ones.

Reply:
> Maybe we can add metadata: NSDictionary field even if Matomo does not support at the moment. We can use known dictionary keys like: count and let the possibility to set custom ones.

count = units here, which is the genetic term that matches dollars, which is the metric you find in every analytics tools.

This is the plan to have a `metadata` or `context` dict but let's see first how count.ly behaves with its `context` while integrating it.